### ISSUE TITLE
message.sendSms - TypeError: Cannot read property 'status' of undefined

### ISSUE THREAD

**Issue by GoEagles999** [Vonage/vonage-node-sdk#188]
I am trying to send an SMS message like this:

```
const from = 'num'
const to = 'num'
const text = 'A text message sent using the Nexmo SMS API'
nexmo.message.sendSms(from, to, text, (error, response) => {
  if(error) {
    throw error;
  } else if(response.messages[0].status != '0') {
    console.error(response);
    throw 'Nexmo returned back a non-zero status';
  } else {
    console.log(response);
  }
});
{ 'message-count': '1',
  messages:
   [ { to: 'num',
       'message-id': '0C000000B7BAC3CE',
       status: '0',
       'remaining-balance': '11.90490000',
       'message-price': '0.03330000',
       network: '23420' } ] }
/srv/node_modules/nexmo/lib/index.js:142
      if (!err && apiResponse.status && apiResponse.messages[0].status > 0) {
                              ^
TypeError: Cannot read property 'status' of undefined
```

As you can see the message successfully sends (verified on my phone) however this error is thrown from the library immediately after. Why is this happening?

**Comment by sebastien-cleany** [comment]
Hello,
I got the same issue. It occurs only with node >= 1.2.0.
Thus I supposed, it's related to https://github.com/nodejs/node/pull/20075

I would suggest the following PR in HttpClient.js but I'm not sure
```js
       response.on("close", e => {
+        if (!e) {
+          return;
+        }
         this.logger.error(
           "problem with API request detailed stacktrace below "
         );
```
I simple workaround is to downgrade Node version to 10.1.0 or below.
I hope this will help you

**Comment by AlexLakatos** [comment]
Thanks @sebastien-cleany , the proposed fix for this is in issue #191 , will be released in version 2.3.1
